### ISSUE TITLE
Add handling of DB-level constraints

### ISSUE THREAD

**Issue by frankdugan3** [ash-project/ash_postgres#56]
I leverage Postgres constraints quite a bit, e.g:
```elixir
create constraint(:attendance_record_types, :shortcut_must_be_lowercase,
  check: "lower(shortcut) = shortcut"
)
```

In Ecto, I'd just use check_constraint so that if the DB kicks back, I can cleanly handle the error rather than throwing an exception: 
```elixir
|> check_constraint(:shortcut,
  name: :shortcut_must_be_lowercase,
  message: "Shortcut keys must be lower-case."
)
```

I would like to be able to specify such constraints and the error messages they should produce.

This may make sense in the `attributes` section:

```elixir
attributes do
  # Tuple of constraint name and message. Could also use keyword syntax for message in case other options are added.
  attribute :shortcut, :string, pg_constraints: [{:shortcut_must_be_lowercase, "Shortcut keys must be lower-case."}]
end
```

More complex constraints may apply to more than one field, the above syntax would allow the same constraint to have different messages on other attributes.

Alternatively, since this would be Postgres-specific, it may be better suited in the `postgres` section of the resource.

**Comment by zachdaniel** [comment]
Yeah, at this point extensions are only able to add custom options inside their dedicated DSL block, so we wouldn't add an option like pg_constraints. But I'll be able to add postgres constraint support easily, so I'll let you know once it's done

**Comment by frankdugan3** [comment]
Resolved with `v0.37.0` release.
